src/Perlang.Tests.Integration/Operator/ShiftOperators.cs

```diff
@@ -0,0 +1,201 @@
+            {
+                string source = @"
+                    print 1 << ""foo"";
+                ";
+
+                var result = EvalWithValidationErrorCatch(source);
+                var exception = result.Errors.FirstOrDefault();
+
+                Assert.Single(result.Errors);
+                Assert.Matches("Unsupported << operands specified: int and string", exception.Message);
+            }
+        }
+
+        public class ShiftRightOperator
+        {
+            [Fact]
+            public void with_integers_performs_right_shifting()
+            {
+                string source = @"
+                    print 65536 >> 6;
+                ";
+
+                string result = EvalReturningOutputString(source);
+                Assert.Equal("1024", result);
+            }
+
+            [Fact]
+            public void takes_precedence_over_multiplication()
+            {
+                string source = @"
+                    print 65536 >> 6 * 3;
+                ";
+
+                string result = EvalReturningOutputString(source);
+                Assert.Equal("3072", result);
+            }
+
+            [Fact]
+            public void takes_precedence_over_division()
+            {
+                string source = @"
+                    print 65536 >> 6 / 3;
+                ";
+
+                // Since the operands are integers, the division is expected to be truncated to its integer portion.
+                // (1024 / 3 = 341.333...)
+                string result = EvalReturningOutputString(source);
+                Assert.Equal("341", result);
+            }
+
+            [Fact]
+            public void takes_precedence_over_addition()
+            {
+                string source = @"
+                    print 65536 >> 6 + 3;
+                ";
+
+                string result = EvalReturningOutputString(source);
+                Assert.Equal("1027", result);
+            }
+
+            [Fact]
+            public void takes_precedence_over_subtraction()
+            {
+                string source = @"
+                    print 65536 >> 6 - 3;
+                ";
+
+                string result = EvalReturningOutputString(source);
+                Assert.Equal("1021", result);
+            }
+
+            [Fact]
+            public void takes_precedence_over_modulo_operator()
+            {
+                string source = @"
+                    print 65536 >> 6 % 1000;
+                ";
+
+                string result = EvalReturningOutputString(source);
+                Assert.Equal("24", result);
+            }
+
+            [Fact]
+            public void takes_precedence_over_power_operator()
+            {
+                string source = @"
+                    print 1024 >> 8 ** 10;
+                ";
+
+                string result = EvalReturningOutputString(source);
+                Assert.Equal("1048576", result);
+            }
```

Comment:
Some of these like the `%` and `**` examples are arguably a bit weird. _Why would anyone in their right mind want to do something like that?_ On the other hand, these examples are not outright _dangerous, as is the case described in #245 (with `&&` and `||`). As long as we keep the precedence of `<<` and `>>` very high, like we currently do, we should be pretty safe.

While the above example is unlikely, I consider the fact that the user would actually want to do `1024 >> (8 ** 10)` even weirder... :slightly_smiling_face: 

Anyway, we'll live with this being a valid combination of operators now, until proven otherwise by experience.